quantilizer.py

```diff
@@ -237,61 +263,93 @@ def test(env_name, dataset_name='ryan', horizon=None, quantiles=[1.0, .5, .25, .
+			ob_list, ac_list, _, proxy_rew_list, true_rew_list = get_trajectories(pi, env, horizon, n_trajectories, play=render)
 
 			proxy_rews.append(proxy_rew_list)
 			true_rews.append(true_rew_list)
 
 			print("->testing for q={} took {}s".format(quantiles[model_nb], 
 												time.time()-start))
 
-		if not os.path.exists('log/rewards'):
-			os.makedirs('log/rewards')	
-		np.save('log/rewards/{}_{}_{}_{}_true'.format(dataset_name, env_name, framework, seed), true_rews)
-		np.save('log/rewards/{}_{}_{}_{}_proxy'.format(dataset_name, env_name, framework, seed), 
+		if not os.path.exists('log/rewards/{}'.format(path)):
+			os.makedirs('log/rewards/{}'.format(path))	
+		np.save('log/rewards/{}{}_{}_{}_{}_{}_true'.format(path, dataset_name, env_name, framework, aggregate_method, seed), true_rews)
+		np.save('log/rewards/{}{}_{}_{}_{}_{}_proxy'.format(path, dataset_name, env_name, framework, aggregate_method, seed), 
 		proxy_rews)
+		
+		result_list.append([ob_list, ac_list])
+	
+	return result_list
 
-def plot(env_name, dataset_name, seed_min=0, seed_nb=1, framework='sklearn'):
+def plot(env_name, dataset_name, seed_min=0, seed_nb=1, framework='sklearn', quantiles=[1.0, .5, .25, .125], plotstyle=None, aggregate_method='continuous', path=''):
+	tr_list, pr_list = [], []
+	opt_val = [-180.16, -79.79] if env_name == 'MountainCar-v0' else [37.4, 0.603]
+
+	
 	for seed in range(seed_min, seed_min + seed_nb):
-		proxy_rews_list = np.load('log/rewards/{}_{}_{}_{}_proxy.npy'.format(dataset_name, env_name, framework, seed))
-		true_rews_list = np.load('log/rewards/{}_{}_{}_{}_true.npy'.format(dataset_name, env_name, framework, seed))
-		qs = [1.0, .5, .25, .125]
-		opt_val = [-180.16, -79.79] if env_name == 'MountainCar-v0' else [37.4, 0.603]
-		true_rewards = [np.mean([sum(traj) for traj in true_arr]) for true_arr in true_rews_list] + [opt_val[0]]
-		proxy_rewards = [np.mean([sum(traj) for traj in proxy_arr]) for proxy_arr in proxy_rews_list] + [opt_val[1]]
-		graph_one(true_rewards, proxy_rewards, qs, env_name, dataset_name, framework, seed)
+		# loading rewards from testing
+		proxy_filename = 'log/rewards/{}{}_{}_{}_{}_{}_proxy.npy'.format(path, dataset_name, env_name, framework, aggregate_method, seed)
+		true_filename = 'log/rewards/{}{}_{}_{}_{}_{}_true.npy'.format(path, dataset_name, env_name, framework, aggregate_method, seed)
+		proxy_rews_list, true_rews_list = np.load(proxy_filename), np.load(true_filename)
+
+		# printing specific stats
+		tr_imit = [sum(traj) for traj in true_rews_list[0]]
+		pr_imit = [sum(traj) for traj in proxy_rews_list[0]]
+		print('[{} {} {} traj seed {}]: tr={}+/-{} (med={}) and pr={}+/-{} (med={})'.format(aggregate_method, framework, len(tr_imit), seed, int(np.mean(tr_imit)), int(np.std(tr_imit)), int(np.median(tr_imit)), int(100 * np.mean(pr_imit)), int(100 * np.std(pr_imit)), int(100 * np.median(pr_imit))))
+		
+		# book-keeping for the multiple seeds plot
+		if plotstyle == 'median_seeds':
```

Comment:
Hmm, this plot function is very long, with a bunch of repetition, and a bunch of "if mode `a`, then make graph `a`, kind of things". I wonder if it is a bit overengineered, and could just benefit from being split into a "get stats" function, and some separate plotting functions for each plot style. But this is more of a "for next time" sort of thought.

Reply:
Yep, I agree that I don't really like this function either. I tried to do this at the end with `plot_distribution``plot_seeds` and `boxplot`, but for the rest I agree that there is too much list comprehension and repetition.

Thanks for remembering me to design better the code, otherwise I get lazy and add another `if case x then do that` and it becomes unelegant.